### PR TITLE
567 improved arg handling

### LINKED ISSUE #567: --vt_lb_stats_dir not being parsed correctly in EMPIRE on Mutrino
**Describe the bug**
Try to specify a directory using `--vt_lb_stats_dir`

### PR DESCRIPTION
**Pull request by lifflander**
Basically:

- All arguments must be accounted for and correctly parsed.
  This mitigates "oops" cases when arguments are not correctly
  handled by appropriate domain.

  To supply MPI_Init args (still does not address issue of late MPI_Init), use:
    `--vt_mpi_args mpiarg1 mpiarg2..`
  To supply user args (returned up through initialize):
    `-- userarg1 userarg2 ..`
  To switch back to VT args, use `--vt_args`

- Also supports 'auto pass-through' args for historic reasons.
  That is, any arguments before a "--vt_" arg including "--vt_help"
  are automatically classified as a pass-through arguments and any
  "--vt_" or argument will implicitly switch into vt-arg mode.

    `1 2 3 --vt_no_color` is the same as
    `--vt_no_color -- 1 2 3` while
    `--vt_no_color 1 2 3` is invalid (left in vt-arg context where such are invalid).

   Most notable, short vt args like '-c' or '-q' do NOT initiate
   vt-arg mode and will subject to pass-through behavior.
   (Short args have been *removed* to avoid this edge-case.)

- The syntax can also be extended indefinitely (although auto-mode only applies once at the start), for whatever use it may have:
   `1 --vt_no_color -- 2 --vt_args --vt_trace -- 3` is the same as
   `1 2 3 --vt_no_color --vt_trace` and
   `--vt_no_color --vt_trace -- 1 2 3`

   It is recommended that parameters that take values do so with the `--param=value` form.

This addresses some PEBCAK errors locally and provides a base for some future changes with directory arg handling. There is a chance similar issues are accidentally encountered on other systems.

Pass-through arguments (and the program name) are displayed in the banner for clarity.

It is a "breaking" change in that some previously valid parameters invocations are not forbidden, ie. needing to use a `--` to switch out of vt-arg context. However changes required are an artifact of the parameters supplied during invocation (as opposed to requiring code changes).

When in vt-arg context, unknown arguments will *abort the program* with an argument error.

The `--vt_help` (replaces `--help`) parameter is also more useful (and program-aborting).

Fixes #567

### CODE REVIEW

**Comment by pnstickne** [comment]
Oh, hmm, hmm, conflicts what..

**Comment by pnstickne** [comment]
Rebased onto origin/develop.

**Comment by pnstickne** [comment]
Well, Codacy is a [getting it wrong] thing here.. :|

**Comment by lifflander** [comment]
@pnstickne This is my fault, but this has gotten delayed. Do we still want to merge this? Can you rebase it if so?

File: src/vt/configs/arguments/args.cc [hunk 1]
```diff
@@ -561,76 +543,180 @@ static std::unique_ptr<char*[]> new_argv = nullptr;
...
-   * Run the parser!
-   */
-  app.allow_extras(true);
-  try {
-    app.parse(args);
-  } catch (CLI::Error &ex) {
-    return app.exit(ex);
+class VtFormatter : public CLI::Formatter {
+public:
+  std::string make_usage(const CLI::App *, std::string name) const override {
```
Comment (lifflander) [inline src/vt/configs/arguments/args.cc:563]: Codacy Issue found: [Function parameter 'name' should be passed by const reference.](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=4937784)

**Comment by lifflander** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/lifflander/vt/pullRequest?prid=4937784&bid=16327427)

**Pull request merged**